features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/TouchPrivacyManager.kt

```diff
@@ -0,0 +1,72 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.sessionreplay.internal
+
+import android.graphics.Point
+import android.graphics.Rect
+import androidx.annotation.UiThread
+import androidx.annotation.VisibleForTesting
+import com.datadog.android.sessionreplay.TouchPrivacy
+
+internal class TouchPrivacyManager(
+    private val globalTouchPrivacy: TouchPrivacy
+) {
+    // areas on screen where overrides are applied
+    private val currentOverrideAreas = HashMap<Rect, TouchPrivacy>()
+
+    // built during the view traversal and copied to currentOverrideSnapshot at the end
+    private val nextOverrideAreas = HashMap<Rect, TouchPrivacy>()
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❔ question:** ‏ I wonder why do you need to keep 2 HashMaps here. What's the purpose of this approach ? Can we add more explanations here please as it is not very obvious from the review. Tks !!

Reply:
I'll add some additional comments to clarify, but briefly it's for two main reasons. First, because the touch handling is happening in parallel to the traversal so we want to resolve privacy using a complete map of privacy areas. Second, in order not to have to track the lifecycle of the overridden views to know when to remove them from the map. Using this method, all items are replaced on every traversal with the current privacy state.